src/python/pants/backend/project_info/peek.py

```diff
@@ -162,21 +193,88 @@ def describe_ruleset(ruleset: DependencyRuleSet | None) -> tuple[str, ...] | Non
     return ruleset.peek()
 
 
+async def _create_target_alias_to_goals_map() -> dict[str, tuple[str, ...]]:
+    """Returns a mapping from a target alias to the goals that can operate on that target.
+
+    For instance, `pex_binary` would map to `("run", "package")`.
+
+    :return: A mapping from a target alias to the goals that can operate on that target.
+    """
+    # TODO: This is manually curated for now - do we want to show all possible goals?
+    peekable_field_sets: list[type[FieldSet]] = [
+        DeployFieldSet,
+        PackageFieldSet,
+        PublishFieldSet,
+        RunFieldSet,
+        TestFieldSet,
+    ]
+    # TODO: How do we get the goal subsystem from field set? This shouldn't need to be manually maintained
+    field_set_to_goal_map: dict[type[FieldSet], str] = {
```

Comment:
Ditto - can this be extracted at runtime?

Reply:
There is information about what rules are using any given field set type, if this helps?

Example:
<img width="675" alt="image" src="https://github.com/pantsbuild/pants/assets/72965/645b966d-d425-4dcc-be3d-9bb3e413d4cc">